draft-ietf-webpush-encryption.md

```diff
@@ -108,170 +101,277 @@ encryption scheme by applications that use push messaging.  An agent that only
 delivers messages that are properly encrypted strongly encourages the end-to-end
 protection of messages.
 
-A web browser that implements the Web Push API [API] can enforce the use of
+A web browser that implements the Web Push API {{API}} can enforce the use of
 encryption by forwarding only those messages that were properly encrypted.
 
 
 ## Notational Conventions
 
 The words "MUST", "MUST NOT", "SHOULD", and "MAY" are used in this document.
 It's not shouting, when they are capitalized, they have the special meaning
-described in [RFC2119].
+described in {{!RFC2119}}.
 
 
-# Key Generation and Agreement
+# Push Message Encryption Overview {#overview}
+
+Encrypting a push message uses elliptic-curve Diffie-Hellman (ECDH) {{ECDH}} to
+establish a shared secret (see {{dh}}) and a symmetric secret for authentication
+(see {{auth}}).
```

Comment:
While potentially premature, it might be worth noting the curve type we want to use early. It's mentioned a few paragraphs down, but almost in passing.